src/zcl_logger.clas.abap

```diff
@@ -386,6 +392,14 @@ CLASS zcl_logger IMPLEMENTATION.
     IF msg_type->type_kind = cl_abap_typedescr=>typekind_struct1
         OR msg_type->type_kind = cl_abap_typedescr=>typekind_struct2.
 
+      IF msg_type->is_ddic_type( ) = abap_true.
```

Comment:
This looks to me as a different way to identify the obj_to_log structure. It may be better but I'd prefer we stick to only method in the codebase. The best would be to open an issue to propose the change and eventually upgrade the whole method to the new way.